sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,431 @@
+  }
+
+  async deserialize(response: HttpOperationResponse): Promise<HttpOperationResponse> {
+    return deserializeAtomXmlResponse(["TopicName", "SubscriptionName", "RuleName"], response);
+  }
+}
+
+enum SqlParameterType {
+  Integer = "l28:int",
+  String = "l28:string",
+  Long = "l28:long",
+  Date = "l28:date"
+}
+/**
+ * Represents type of SQL `Parameter` in ATOM based management operations
+ */
+export type SqlParameter = {
+  key: string;
+  value: string | number;
+  type: string;
+};
+
+/**
+ * Internal representation of SQL parameter info
+ */
+type RawSqlParameter = {
+  Key: string;
+  Value: any;
+};
+
+/**
+ *  @ignore
+ * Helper utility to retrieve array of `SqlParameter` from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getSqlParametersOrUndefined(value: any): SqlParameter[] | undefined {
+  const parameters: SqlParameter[] = [];
+
+  // Ignore special case as Service Bus treats "" as a valid value for SQL parameters
+  if (typeof value === "string" && value.trim() === "") {
+    return undefined;
+  }
+
+  const jsObject: any = getJSObjectOrUndefined(value);
+  if (jsObject == undefined) {
+    return undefined;
+  }
+
+  try {
+    let rawParameters = jsObject["KeyValueOfstringanyType"];
+    if (rawParameters && rawParameters.length && rawParameters.length > 0) {
+      for (let i = 0; i < rawParameters.length; i++) {
+        parameters.push(buildSqlParameter(rawParameters[i]));
+      }
+    } else {
+      parameters.push(buildSqlParameter(rawParameters));
+    }
+    return parameters;
+  } catch (err) {
+    throw new TypeError(
+      `${JSON.stringify(
+        jsObject,
+        undefined,
+        2
+      )} expected to be an array of Parameter instances, or undefined :: ${err.message}`
+    );
+  }
+}
+
+/**
+ * Helper utility to build an instance of parsed SQL parameteras `Parameter` from given input,
+ * @param value
+ */
+function buildSqlParameter(value: RawSqlParameter): SqlParameter {
+  const rawValue = value["Value"]["_"];
+  const type = value["Value"]["$"]["i:type"].toString().substring(5);
+  let parsedValue: any;
+  switch (type) {
+    case "int":
+      parsedValue = getIntegerOrUndefined(rawValue);
+      break;
+    case "string":
+    case "long":
+    case "date":
+      parsedValue = rawValue;
+      break;
+
+    default:
+      throw new Error(
+        `Invalid type "${type}" on the SQL Parameter. Must be either of "interface, "string", "long" or "date".`
+      );
+  }
+  const parameter: SqlParameter = {
+    key: value["Key"],
+    value: parsedValue,
+    type: type
+  };
+  return parameter;
+}
+
+/**
+ *  @ignore
+ * Helper utility to extract array of `RawSqlParameter` instances from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getRawSqlParameters(parameters: SqlParameter[] | undefined): any {
+  if (parameters == undefined || (parameters && parameters.length && parameters.length == 0)) {
```

Comment:
A better check would be `if (!Array.isArray(parameters))`

Reply:
Updated